added online compiler feature

monikasingh20 opened the pull request:
Fixes issue: 44


Changes:
I have added Online Compiler link which would link to  https://codecompilerun.herokuapp.com/ api . 
But this api supports languages : c, c++,java,python,javascript, c#  only.
These compiler donts have post requests feature , so user have to copy paste the code to compiler in order to run code.


<!-- Make sure that you enjoyed being the part of the OpenGenus Community. We would love to hear how can we make your contributing experience better. Thank you. Have a nice day! -->

monikasingh20 commented:
Sir , please review this and suggest any changes required .

arnavb commented on code/popup.html:
```diff
@@ -31,9 +31,11 @@
-                  <a class="btn float-right" target="_blank" id="help" href="javascript:void(0)">Help</a>
+            <div class="col-sm-5">
+				
+				  <a class="btn float-right" target="_blank" title="Works only when you are online" href="https://codecompilerun.herokuapp.com/">OnlineComplie</a>
```
Looks like you have a typo in the link text.

monikasingh20 replied:
I don't found any typo error. Please check it once again.
still if you find it , please tell in detail.

arnavb replied:
Online _Complie_

monikasingh20 replied:
Ohk I will correct it. 😃

monikasingh20 replied:
I have changed the typo error.please verify it.

AdiChat commented:
Nice work :+1: 

A compiler is useful for our users but at the same time, placing a link will not help much in the sense that it will require an internet connection. 
The initial issue was concerned with integrating a compiler with Quark which would have been different as it would have provided a native solution defaulting to an editor while offline. 

You may work on this issue further to achieve the above objective. The first step would be to provide an editable interface for the code results. :+1:

monikasingh20 commented:
We are unable to run shell commands from js file instead we need nodejs for the same. Please suggest a good option.

AdiChat commented:
Yes, for the extension, we need to rely on external APIs or develop our own API (which should be simple to build). This will be an online feature and should stay disabled while offline. To achieve this, we must move one step at a time.

A good progress would be to develop an editor within Quark where users can create and edit files. Note, that the editor will work offline. Once done, we must move to develop a feature to enable the codes to open up in the editor instead of a new tab with no formatting. Similarly, we will move through further steps. 

You may proceed to develop the editor. :+1:

monikasingh20 commented:
I have added compiler option , clicking on compiler option will open  editor html page, in which user can either choose file and load ,edit the file. Or directly copy file and edit and save in local system.

monikasingh20 commented:
Sir, please review and suggest any changes and what next to be done.

AdiChat commented:
Great work. :+1: 

You may use the term "Editor" instead of "Compiler" for the current state in the button and editor. You may, also, consider increasing the font size within the editor. The position of the control buttons should be better You may place it on the header of the editor as dropdowns in compact form or experiment with other designs. 

Kindly take a look at the above points and make appropriate changes. :+1:

monikasingh20 commented:
@AdiChat I have done few changes:
1) replaced compiler with Editor name.
2) Improved UI design of Editor.
3) Added features like zoomin, zoomout, clear,undo, save,upload code.
4) used dropdown menu.

![screenshot from 2018-03-11 23-32-23](https://user-images.githubusercontent.com/29702439/37256723-b30d509c-2584-11e8-832c-78ff79299414.png)
![screenshot from 2018-03-11 23-32-33](https://user-images.githubusercontent.com/29702439/37256731-c2a0540a-2584-11e8-8857-6466004261f2.png)

monikasingh20 commented:
@AdiChat please review above changes.

AdiChat commented on code/edit.html:
```diff
@@ -0,0 +1,64 @@
+    </tr>
+</table>
+ 
+<script type="text/javascript">
```
You need to separate `JS` code from `HTML` in case of extensions.

AdiChat commented on code/edit.html:
```diff
@@ -0,0 +1,64 @@
+    <tr>
+        <td>Select a File to Load:</td>
+        <td><input type="file" id="fileToLoad"></td>
+        <td><button onclick="loadFileAsText()">Load Selected File</button><td>
```
`onclick` does not work in extensions. You can trigger the `loadFileAsText` function by a `js` code listening for a click event on the button.

AdiChat commented:
Great work :+1:

The text typed in the editor must be saved locally by default. You must provide a clear option as well. Also, the OpenGenus logo must appear at the top left corner. 

In the editor.html, the header should be `Editor` instead of `Edit your code`.

I have added some comments as well. 

Kindly take a look into this and make appropriate changes. :+1:
